### ISSUE TITLE
[Feature Request] Image Augmentation Is All You Need benchmark

### ISSUE THREAD

**Issue by vmoens** [pytorch/rl#32]
Nice benchmark to have as it would interact with other torch libraries

http://arxiv.org/abs/2004.13649

**Comment by yemaedahrav** [comment]
Hi, I would be interested in taking up this issue. Should I assign myself on this?

**Comment by vmoens** [comment]
That would be lovely!
From what I understood the main technical difficulty is to do transforms in a consistent manner (eg, have an action mapping operator that follows the image/state transform etc).
Not sure what form these will take but happy to help or provide some guidance if needed. 
If you want to discuss this not on gh, feel free to join our slack channel or discord (but gh is also perfectly fine).

**Comment by yemaedahrav** [comment]
I did go through the [/benchmarks](https://github.com/pytorch/rl/tree/main/benchmarks) directory (including the commit history) and take a look at the code there and had read the paper as well. I feel I would still require more clarity on the goals this issue attempts to accomplish. Is it to get a working implementation integrated into the TorchRL library or am I missing something?

**Comment by vmoens** [comment]
I would say that the primary goal is that if anyone wants to build on top of this work a new research paper or an application (eg apply it to a different domain or else) it should be doable.
The secondary goals are:
- provide the set of primitives that are needed for this. In this case I think these will be transforms to append to the environment.
- provide a script that reproduces some of the paper's result to showcase how those primitives should be used.

This hierarchy of values is important to keep in mind while implementing all this: we don't want to write a new benchmark for the sake of the script, we want the benchmark to show how to use what that paper proposes using torchrl.

I hope that makes sense!

**Comment by neo-alex** [comment]
I am looking forward to this DrQ implementation!

Just wondering if it wouldn't be better to rather reproduce DrQv2 directly? It should be the same amount of work but the performance is vastly superior, as shown in the first illustration of the paper: https://arxiv.org/abs/2107.09645 (the image augmentation trick is the same but the v2 made a couple of changes wrt. hyperparameters, replacing SAC with DDPG...).

By the way, you can find the official implementation of DrQv2 here as an inspiration:
https://github.com/facebookresearch/drqv2

**Comment by vmoens** [comment]
That makes sense yes!

Denis (the author of the repo) isn't at meta anymore but we could still reach out if we need help!

**Comment by yemaedahrav** [comment]
Sounds good, is there any other such benchmark we have in this torchrl library that I can refer to?

**Comment by vmoens** [comment]
Not sure no. You can always check our examples in the examples folder.

**Comment by PaulBeuran** [comment]
Is this issue still open ? I'd like to contribute

**Comment by yemaedahrav** [comment]
Yes, I haven't really gotten around to completing this, however I did read up and make some progress. Maybe you could be a co-assignee on this issue and we can work together?

**Comment by PaulBeuran** [comment]
That would be great, thanks. 
Could we exchange through DIscord on the subject ? 
DM @pbeuran_71003 if that's the case.

**Comment by alektebel** [comment]
Hi, I've noticed the discussion about implementing the DrQ benchmark stalled about a year ago.
I'd like to help move this forward! @yemaedahrav , @PaulBeuran , I'd be happy to co-author this. Can you add me to discord so that I can see the current state of the PR and look into what I can help?
Thanks
